### PR TITLE
New SDK feature disablers + USB disabler fix with DEBUG builds

### PR DESCRIPTION
**Pull request by apaillier-ledger**
## Description

* Building an app with disabled USB would not work with DEBUG builds since on devices other that the Nano S, the printf implementation uses the USB
* The UI (NBGL or BAGL) can be completely disabled for faster compilation when not needed (Ethereum plugins for example)
* Fixed builds with seproxyhal not enabled + added a disabler for it in the standard_app Makefile
* Breaking change for Stax apps that use the QR code feature without explicitly adding its source path

## Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Tests
- [ ] Documentation
- [ ] Other (for changes that might not fit in any category)

### CODE REVIEW

File: Makefile.defines [hunk 1]
```diff
@@ -172,24 +172,7 @@ LDFLAGS  += -L$(BOLOS_SDK)/arch/st33k1/lib/
...
-DEFINES += HAVE_BAGL_ELLIPSIS
-DEFINES += HAVE_BAGL_FONT_OPEN_SANS_REGULAR_11PX
-DEFINES += HAVE_BAGL_FONT_OPEN_SANS_EXTRABOLD_11PX
-DEFINES += HAVE_BAGL_FONT_OPEN_SANS_LIGHT_16PX
-
-LDFLAGS  += -L$(BOLOS_SDK)/lib/
-endif
-
+ifeq ($(NO_UI), 0)
 ifeq ($(TARGET_NAME),TARGET_STAX)
```
Comment (xchapron-ledger) [inline Makefile.defines:176]: why keeping this defines when in `NO_UI`?

File: src/os_io_task.c [hunk 2]
```diff
@@ -33,18 +33,26 @@ void io_start(void) {
 void bolos_init_ios(void) {
```
Comment (xchapron-ledger) [inline src/os_io_task.c:33]: I' didn't know this exists... this is somehow a duplicate of what's done in lib_standard_app/main.c: common_app_init(). Maybe we can merge?

File: Makefile.standard_app [hunk 3]
```diff
@@ -105,10 +111,8 @@ ifneq ($(DISABLE_STANDARD_WEBUSB), 1)
     DEFINES += HAVE_WEBUSB WEBUSB_URL_SIZE_B=$(WEBUSB_URL_SIZE_B) WEBUSB_URL=$(WEBUSB_URL)
 endif
 
-ifneq ($(DISABLE_STANDARD_BAGL_UX_FLOW), 1)
-ifneq ($(TARGET_NAME), TARGET_STAX)
-    DEFINES += HAVE_UX_FLOW
```
Comment (xchapron-ledger) [inline Makefile.standard_app:110]: I' think this one is useful for some apps on Nanos (but should not be put on all apps).

**Review by xchapron-ledger — COMMENTED** [review]
CI is failing on many apps :/

File: Makefile.defines (hunk 1 above)
Comment (xchapron-ledger) [reply Makefile.defines:177]: @apaillier-ledger Why is this necessary?

File: Makefile.defines [hunk 4]
```diff
@@ -201,6 +185,22 @@ DEFINES += HAVE_SE_TOUCH
...
+endif
+ifeq ($(TARGET_NAME),$(filter $(TARGET_NAME),TARGET_NANOX TARGET_NANOS2))
+DEFINES += HAVE_UX_FLOW
+DEFINES += BAGL_HEIGHT=64
+DEFINES += HAVE_BAGL_ELLIPSIS
+DEFINES += HAVE_BAGL_FONT_OPEN_SANS_REGULAR_11PX
+DEFINES += HAVE_BAGL_FONT_OPEN_SANS_EXTRABOLD_11PX
+DEFINES += HAVE_BAGL_FONT_OPEN_SANS_LIGHT_16PX
+endif
+LDFLAGS  += -L$(BOLOS_SDK)/lib/
```
Comment (xchapron-ledger) [inline Makefile.defines:202]: What is this? A remaining from a very very old SDK?

**Review by xchapron-ledger — APPROVED** [review]

**Review by nroggeman-ledger — APPROVED** [review]

**Pull request merged**